tests/test_mc_comm.py

```diff
@@ -0,0 +1,229 @@
+    simulator = mantarray_mc_simulator_no_beacon[4]
+
+    board_idx = 0
+    invoke_process_run_and_check_errors(mc_process)
+    assert mc_process.is_registered_with_serial_comm(board_idx) is False
+
+    mc_process.set_board_connection(board_idx, simulator)
+    test_bytes = SERIAL_COMM_MAGIC_WORD_BYTES[3:] + bytes(8)
+    test_bytes += create_data_packet(
+        0,
+        SERIAL_COMM_MAIN_MODULE_ID,
+        SERIAL_COMM_STATUS_BEACON_PACKET_TYPE,
+        DEFAULT_SIMULATOR_STATUS_CODE,
+    )
+    test_item = {"command": "add_read_bytes", "read_bytes": test_bytes}
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(
+        test_item, testing_queue
+    )
+    invoke_process_run_and_check_errors(simulator)
+
+    invoke_process_run_and_check_errors(mc_process)
+    assert mc_process.is_registered_with_serial_comm(board_idx) is True
+    # make sure no errors in next iteration
+    invoke_process_run_and_check_errors(mc_process)
+
+
+def test_McCommunicationProcess_register_magic_word__registers_magic_word_in_serial_comm_from_board__when_first_packet_is_not_truncated(
+    four_board_mc_comm_process, mantarray_mc_simulator_no_beacon, mocker
+):
+    mc_process = four_board_mc_comm_process[0]
+    testing_queue = mantarray_mc_simulator_no_beacon[3]
+    simulator = mantarray_mc_simulator_no_beacon[4]
+
+    board_idx = 0
+    invoke_process_run_and_check_errors(mc_process)
+    assert mc_process.is_registered_with_serial_comm(board_idx) is False
+
+    mc_process.set_board_connection(board_idx, simulator)
+    test_bytes = create_data_packet(
+        0,
+        SERIAL_COMM_MAIN_MODULE_ID,
+        SERIAL_COMM_STATUS_BEACON_PACKET_TYPE,
+        DEFAULT_SIMULATOR_STATUS_CODE,
+    )
+    test_item = {"command": "add_read_bytes", "read_bytes": test_bytes}
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(
+        test_item, testing_queue
+    )
+    invoke_process_run_and_check_errors(simulator)
+
+    invoke_process_run_and_check_errors(mc_process)
+    assert mc_process.is_registered_with_serial_comm(board_idx) is True
+    # make sure no errors in next iteration
+    invoke_process_run_and_check_errors(mc_process)
+
+
+def test_McCommunicationProcess_register_magic_word__registers_with_magic_word_in_serial_comm_from_board__when_first_packet_is_truncated_to_less_than_8_bytes__and_calls_read_with_correct_size__and_calls_sleep_correctly(
+    four_board_mc_comm_process, mantarray_mc_simulator_no_beacon, mocker
+):
+    # mock sleep to speed up the test
+    mocked_sleep = mocker.patch.object(mc_comm, "sleep", autospec=True)
+
+    mc_process = four_board_mc_comm_process[0]
+    simulator = mantarray_mc_simulator_no_beacon[4]
+
+    # need to mock read here to have better control over the reads going into McComm
+    test_read_values = [
+        SERIAL_COMM_MAGIC_WORD_BYTES[:4],
+        bytes(0),
+        bytes(0),
+        bytes(0),
+        bytes(0),
+        SERIAL_COMM_MAGIC_WORD_BYTES[4:],
```

Comment:
why are you using `4`? is it arbitrary, or half the size of the word? is it `length of magic word minus the amount truncated 5 lines earlier`? a comment might be useful